routers/api/v1/api.go

```diff
@@ -1206,7 +1218,7 @@ func Routes(ctx gocontext.Context) *web.Route {
 				m.Post("/{username}/{reponame}", admin.AdoptRepository)
 				m.Delete("/{username}/{reponame}", admin.DeleteUnadoptedRepository)
 			})
-		}, reqToken(auth_model.AccessTokenScopeSudo), reqSiteAdmin())
+		}, reqTokenOrSiteAdmin(auth_model.AccessTokenScopeSudo))
```

Comment:
Maybe we could introduce a `Or` function here to composite these two middlewares but not write a new one.

Reply:
That would be not trivial, since the existing functions calls `ctx.Error` directly. So we can't just call one, and if fails call another. The way error is returned must be changed then.